lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -40,19 +41,48 @@ export default function AssignmentActivity() {
   }>();
 
   const { filters, updateFilters, urlWithFilters } = useDashboardFilters();
-  const { studentIds } = filters;
+  const { studentIds, segmentIds } = filters;
   const search = useSearch();
   const [, navigate] = useLocation();
 
   const assignment = useAPIFetch<AssignmentDetails>(
     replaceURLParams(routes.assignment, { assignment_id: assignmentId }),
   );
   const autoGradingEnabled = !!assignment.data?.auto_grading_config;
+  const segments = useMemo((): DashboardActivityFiltersProps['segments'] => {
+    const { data } = assignment;
+    // For now, we want to display the segments filter only for auto-grading
+    // assignments, but this will eventually change
+    if (!data || !autoGradingEnabled) {
+      return undefined;
+    }
+
+    const hasSections = 'sections' in data;
+    const entries = hasSections
+      ? data.sections
+      : 'groups' in data
+        ? data.groups
+        : undefined;
+
+    // If the assignment doesn't have neither sections nor groups, we won't
```

Comment:
```suggestion
    // If the assignment doesn't have either sections or groups, we won't
```

The comment is easier to parse without the double-negative of "doesn't" and "neither"/"nor".